pandas/core/groupby/ops.py

```diff
@@ -606,9 +615,49 @@ def _ea_wrap_cython_operation(
             f"function is not implemented for this dtype: {values.dtype}"
         )
 
+    @final
+    def _masked_ea_wrap_cython_operation(
+        self,
+        kind: str,
+        values: BaseMaskedArray,
+        how: str,
+        axis: int,
+        min_count: int = -1,
+        **kwargs,
+    ) -> BaseMaskedArray:
+        """
+        Equivalent of `_ea_wrap_cython_operation`, but optimized for masked EA's
+        and cython algorithms which accept a mask.
+        """
+        orig_values = values
+
+        # Copy to ensure input and result masks don't end up shared
+        mask = values._mask.copy()
+        arr = values._data
+
+        if is_integer_dtype(arr.dtype) or is_bool_dtype(arr.dtype):
+            # IntegerArray or BooleanArray
+            arr = ensure_int_or_float(arr)
```

Comment:
FWIW im planning to kill off this function; for EAs this is always just `arr.to_numpy(dtype="float64", na_value=np.nan)`

Reply:
Thanks for bringing up - realized this whole condition can be simplified since we actually have an ndarray at this point